crates/proto/src/op/header.rs

```diff
@@ -514,7 +514,7 @@ impl BinEncodable for Header {
         q_opcd_a_t_r |= u8::from(self.op_code) << 3;
         q_opcd_a_t_r |= if self.authoritative { 0x4 } else { 0x0 };
         q_opcd_a_t_r |= if self.truncation { 0x2 } else { 0x0 };
-        q_opcd_a_t_r |= if self.recursion_desired { 0x1 } else { 0x0 };
+        q_opcd_a_t_r |= u8::from(self.recursion_desired); // 0x1 or 0x0
```

Comment:
hm, I'm not sure I agree with this change. This requires knowledge that boolean to u8 conversion is absolutely `0x0`or `0x1`, which in my opinion is non-obvious.

Reply:
From a system compiler background perspective it's OK, but I suppose it's not for everyone, so I agree it's a per-project choice. At least it's settled. :)